src/hooks/use-require-auth.ts

```diff
@@ -0,0 +1,55 @@
+"use client";
+
+import * as React from "react";
+import { useSession } from "@/contexts/rainbowkit-siwe-iron-session-provider";
+import { SessionData } from "@/lib/session/types";
+import { usePathname, useRouter } from "next/navigation";
+
+const useRequireAuth = ({ enabled }: { enabled?: boolean }) => {
+  const pathname = usePathname();
+  const router = useRouter();
+  const { status, data: session } = useSession();
+
+  const isHomePage = pathname === "/";
+  const isClaimPage = pathname.startsWith("/claim");
+  const isAllocationPage = pathname.startsWith("/allocation");
+  const hasSession = status === "authenticated";
+
+  React.useEffect(() => {
```

Comment:
Could we perhaps reference things directly instead of importing the React class in this way.
```
import { useEffect } from "react";
```

It's my understanding this loads less, however that might not be the case these days

Reply:
We can do this. it doesn't have an effect on loading I think react is loaded into all jsx files. I use react exports like this so it's clear what's coming from react and what's not but I don't feel super strongly.

Reply:
Ahh ok, Its just more consistent with the rest of the importing approaches throughout the code base